biovectors_modules/word2vec_run_helper.py

```diff
@@ -1,62 +1,165 @@
 import gzip
+from pathlib import Path
 import random
+import tarfile
 
+import lxml.etree as ET
 import pandas as pd
+import spacy
 
+nlp = spacy.load("en_core_web_sm")
```

Comment:
Will you always use `nlp` when this file is imported? Iirc running `spacy.load` takes a couple seconds and this line will be called whenever the module is imported. It may be better to put it in the `PubMedSentencesIterator` class's init function